processing/src/main/java/org/apache/druid/java/util/common/StringUtils.java

```diff
@@ -373,16 +373,24 @@ public static String nonStrictFormat(String message, Object... formatArgs)
     if (formatArgs == null || formatArgs.length == 0) {
       return message;
     }
+    if (message == null) {
```

Comment:
Instead of these changes, it would be simpler to just do a `Preconditions.notNull(message)` at the start of this method.

Reply:
Preconditions check will throw an exception again.   Some of the exceptions are not having message at all and we are calling nonStrictFormat while making DruidException, esp exception from third party library. As the name says, expectation from nonStrictFormat in this case is that it should just give back some non null string. Best case effort is to append the args to string and give it back.

If we add precondition then we need to safeguard the invoker lines with not null.

Reply:
Based on comments, I found the instances where we could potentially pass null as formatString. I fixed all of them I could find in this PR. I am still doubtful about adding precondition check to nonStrictFormat as it is used at tons of places. I removed the change here and fixing the invokers.